homeassistant/components/modbus/validators.py

```diff
@@ -60,6 +62,7 @@
         "slave_count",
         "swap_byte",
         "swap_word",
+        "allow_register_size_set",
```

Comment:
This should always be set, so no need to have an extra variable.

Reply:
This was for the case of string as structure ">s" would still be array of 1byte characters

Reply:
Then you could check "datatype: string", the code becomes leaner when register_size is always set (no strange "if")